lib/src/main/java/org/team100/lib/motion/ElevatorArmWristKinematics.java

```diff
@@ -36,19 +37,34 @@ public Pose2d forward(Config config) {
         return new Pose2d(x, y, new Rotation2d(r));
     }
 
+
     public Config inverse(Pose2d pose) {
         Translation2d wristPosition = pose.getTranslation()
                 .minus(new Translation2d(m_manipulatorLength, pose.getRotation()));
-        // always reach up
-        // TODO: change this, sometimes we have to reach down.
+
+        double direction, wristAngle;
+        if (pose.getX()<m_armCrossoverHeight) { //should be reaching down here
+                direction = -1.0;
+        } else {
+                direction = 1.0;
+        }
+
         Translation2d wristToShoulder = new Translation2d(
-                Math.sqrt(m_armLength * m_armLength - wristPosition.getY() * wristPosition.getY()),
-                wristPosition.getY());
-        Translation2d shoulderPosition = wristPosition.minus(wristToShoulder);
+                (direction*(Math.sqrt(m_armLength * m_armLength - wristPosition.getY() * wristPosition.getY()))),
+                wristPosition.getY()); //the minus in sqrt is to flip arm to facing down
+                System.out.println(wristToShoulder);
+        Translation2d shoulderPosition = wristPosition.minus(wristToShoulder); 
```

Comment:
i think this code is confusing because of all the "minus" operations. if the translation2d represents the translation *from* the wrist *to* the shoulder, then the "reach down" case should have a *positive* x value.  also, below, using the angle of this "from wrist to shoulder" isn't right, you want the angle "from shoulder to wrist": the end-effector pose is just the sum of "shoulder to wrist" and "wrist to tip".  so i'd suggest redoing all these minus signs so there are fewer of them, and make this translation represent "shoulder to wrist".  it's the "normal" thing in robotics for angles to be measured starting at the "body" and ending at the "finger" so we should do that too.

Reply:
one more suggestion: pull the expression inside "sqrt()" into a local variable, and do something when that variable is negative -- maybe override it with zero.  that would mean if you attempt to reach outside the work envelope, the resulting config is as close to that as it can go (arm horizontal), instead of just failing.